cmake/WASINNDeps.cmake

```diff
@@ -567,3 +573,97 @@ function(wasmedge_setup_mlx_target target)
     )
   endif()
 endfunction()
+
+function(wasmedge_setup_bitnet_target target)
+  if(NOT TARGET llama)
+
+    set(BUILD_SHARED_LIBS OFF)
+    set(LLAMA_BUILD_TESTS OFF)
+    set(LLAMA_BUILD_EXAMPLES OFF)
+    set(LLAMA_BUILD_SERVER OFF)
+    set(LLAMA_ALL_WARNINGS OFF)
+    set(LLAMA_CURL OFF)
+    set(GGML_ACCELERATE OFF)
+    set(GGML_BLAS OFF)
+    set(GGML_OPENMP OFF)
+   
+    if(WASMEDGE_PLUGIN_WASI_NN_BITNET_ARM_TL1)
+      message(STATUS "WASI-NN BitNet backend: Enable ARM TL1")
+      set(BITNET_ARM_TL1 ON)
+    endif()
+    if(WASMEDGE_PLUGIN_WASI_NN_BITNET_X86_TL2)
+      message(STATUS "WASI-NN BitNet backend: Enable x86 TL2")
+      set(BITNET_X86_TL2 ON)
+    endif()
+    if(WASMEDGE_PLUGIN_WASI_NN_BITNET_ARM_TL1 AND WASMEDGE_PLUGIN_WASI_NN_BITNET_X86_TL2)
+    message(FATAL_ERROR "WASI-NN BitNet backend: Both ARM_TL1 or X86_TL2 optimizations cannot be enabled simultaenously")
+  endif()
```

Comment:
Maybe you should format the code; this seems to be at the same level as line 598.